conn.go

```diff
@@ -1430,6 +1612,34 @@ func (c *Conn) executeQuery(ctx context.Context, qry *Query) *Iter {
 	case *resultVoidFrame:
 		return &Iter{framer: framer}
 	case *resultRowsFrame:
+		if x.meta.newMetadataID != nil {
+			// Updating the result metadata id in prepared stmt
+			//
+			// If a RESULT/Rows message reports
+			//      changed resultset metadata with the Metadata_changed flag, the reported new
+			//      resultset metadata must be used in subsequent executions
+
+			stmtCacheKey := c.session.stmtsLRU.keyFor(c.host.HostID(), c.currentKeyspace, qry.stmt)
+			oldInflight, ok := c.session.stmtsLRU.get(stmtCacheKey)
+			if !ok {
+				// We didn't find the stmt in the cache, so we just re-prepare it
+				return c.executeQuery(ctx, qry)
+			}
+
+			newInflight := &inflightPrepare{
+				done: make(chan struct{}),
+				preparedStatment: &preparedStatment{
+					id:               oldInflight.preparedStatment.id,
+					resultMetadataID: x.meta.newMetadataID,
+					request:          oldInflight.preparedStatment.request,
+					response:         x.meta,
+				},
+			}
```

Comment:
This causes a deadlock because the `done` channel will never be closed so the goroutine will be stuck trying to read from it.
I think you're already in the process of fixing this flow according to [your comment here](https://github.com/apache/cassandra-gocql-driver/pull/1822/files#r1817022382) but I did find this so I figured I'd point it out regardless.

Reply:
Oh... I forgot to close this `done` after adding a new inflight to the cache... But it will look a bit weird to me... However, using the old one `done` here is not a solution because there won't be a guarantee that the updated metadata is used by other goroutines.

Reply:
Yeah it's fine to create the channel and close it right away, it might look a bit weird but it's the only way to make this work unless we change how the ps cache works.